contracts/eos/verifier/src/cpp/nativeverifier/nativeverifier.cpp

```diff
@@ -0,0 +1,256 @@
+#include <eosio/eosio.hpp>
+#include <eosio/crypto.hpp>
+#include <groth16/groth16.hpp>
+#include <groth16/rapiduint256.hpp>
+
+using namespace eosio;
+
+static constexpr uint32_t ROOT_LENGTH = 32;
+static constexpr uint32_t PROOF_LENGTH = 384;
+
+static const char hex_chars[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+
+class [[eosio::contract("nativeverifier")]] nativeverifier : public eosio::contract
```

Comment:
```suggestion
class [[eosio::contract("verifiernative")]] verifiernative : public eosio::contract
```
`verifiernative` seems like a better name?